kraken/src/modules/attacks/dns_resolution.rs

```diff
@@ -0,0 +1,137 @@
+                        destination = cname_rec.to;
+                        dns_record_type = DnsRecordType::Cname;
+                    }
+                    Record::Caa(caa_rec) => {
+                        source = caa_rec.source;
+                        destination = caa_rec.to;
+                        dns_record_type = DnsRecordType::Caa;
+                    }
+                    Record::Mx(mx_rec) => {
+                        source = mx_rec.source;
+                        destination = mx_rec.to;
+                        dns_record_type = DnsRecordType::Mx;
+                    }
+                    Record::Tlsa(tlsa_rec) => {
+                        source = tlsa_rec.source;
+                        destination = tlsa_rec.to;
+                        dns_record_type = DnsRecordType::Tlsa;
+                    }
+                    Record::Txt(txt_rec) => {
+                        source = txt_rec.source;
+                        destination = txt_rec.to;
+                        dns_record_type = DnsRecordType::Txt;
+                    }
+                };
+
+                self.send_ws(WsMessage::DnsResolutionResult {
+                    attack_uuid: self.attack_uuid,
+                    source: source.clone(),
+                    destination: destination.clone(),
+                })
+                .await;
+
+                self.insert_dns_result(source, destination, dns_record_type)
+                    .await?;
+
+                Ok(())
+            },
+        )
+        .await;
+
+        self.set_finished(result.err()).await;
+    }
+
+    /// Insert a dns resolution result and update the aggregation
+    async fn insert_dns_result(
+        &self,
+        source: String,
+        destination: String,
+        dns_record_type: DnsRecordType,
+    ) -> Result<(), rorm::Error> {
+        let mut tx = self.db.start_transaction().await?;
+
+        if query!(&mut tx, DnsResolutionResult)
+            .condition(and!(
+                DnsResolutionResult::F.attack.equals(self.attack_uuid),
+                DnsResolutionResult::F
+                    .dns_record_type
+                    .equals(dns_record_type.clone()),
+                DnsResolutionResult::F.source.equals(&source),
+                DnsResolutionResult::F.destination.equals(&destination)
+            ))
+            .optional()
+            .await?
+            .is_some()
+        {
+            debug!("entry already exists");
+        } else {
+            insert!(&mut tx, DnsResolutionResult)
+                .single(&DnsResolutionResultInsert {
+                    uuid: Uuid::new_v4(),
+                    attack: ForeignModelByField::Key(self.attack_uuid),
+                    dns_record_type,
+                    source: source.clone(),
+                    destination: destination.clone(),
+                })
+                .await?;
+
+            Domain::insert_if_missing(&mut tx, self.workspace_uuid, &source).await?;
```

Comment:
If inserting a cname, shouldn't be both source and target be added as domains?
The same for A, AAAA, the Host insert is missing